pkg/util/modules/modules.go

```diff
@@ -67,41 +75,46 @@ func (m *Manager) AddDependency(name string, dependsOn ...string) error {
 // InitModuleServices initialises the target module by initialising all its dependencies
 // in the right order. Modules are wrapped in such a way that they start after their
 // dependencies have been started and stop before their dependencies are stopped.
-func (m *Manager) InitModuleServices(target string) (map[string]services.Service, error) {
-	if _, ok := m.modules[target]; !ok {
-		return nil, fmt.Errorf("unrecognised module name: %s", target)
+func (m *Manager) InitModuleServices(name string) error {
```

Comment:
Could we make it work with this signature instead?

```suggestion
func (m *Manager) InitModuleServices(names ...string) (map[string]services.Service, error)
```

There are multiple projects using this system, and this is backwards compatible, while your suggestion isn't. Furthermore it keeps the state of initialization outside of this manager.

Reply:
Sure, didn't know that anything else other than Cortex used it.